charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -2,19 +2,80 @@
 Return the name of the configMap holding the Super Agent's config. Defaults to release's fill name suffiexed with "-config"
 */ -}}
 {{- define "newrelic-super-agent.config.name" -}}
-{{- (include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" )) -}}
+{{- include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" "local-data" "suffix" "superagent-config" ) -}}
 {{- end -}}
 
+
+
+{{- /*
+Return the agents part that should go in the super agent config. It is created from `.Values.config.subAgents`.
+*/ -}}
+{{- define "newrelic-super-agent.config.agents.yaml" -}}
+{{- if (.Values.config).subAgents -}}
+{{- $agents := dict -}}
+{{- range $subAgentName, $subAgentConfig := (.Values.config).subAgents -}}
```

Comment:
why the () in  `(.Values.config).subAgents`?

Reply:
If `.Values.config` is `nil` and you try to access `subAgents` if panics and does not template.

But if you use `(.Values.config)` it returns "empty object" so you can access safely to `subAgents` that will return `nil`.

It is a safe mechanism as we cannot use [`dig` filter](http://masterminds.github.io/sprig/dicts.html#dig).

Reply:
Thanks it is better!